api-put-object.go

```diff
@@ -170,6 +170,33 @@ func (c Client) putObjectNoChecksum(bucketName, objectName string, reader io.Rea
 		return 0, ErrEntityTooLarge(size, maxSinglePutObjectSize, bucketName, objectName)
 	}
 
+	if size <= -1 {
+		// Initialize a new temporary file.
+		var tmpFile *tempFile
+		tmpFile, err = newTempFile("single$-putobject-single")
+		if err != nil {
+			return 0, err
+		}
+		defer tmpFile.Close()
+		size, err = io.Copy(tmpFile, io.LimitReader(reader, maxSinglePutObjectSize))
+		if err != nil {
+			return 0, err
+		}
+		if size > maxSinglePutObjectSize {
```

Comment:
Yes this is a wrong check. In-fact we can set the size to the max value and let the underlying reader be sent to the server.